Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -30,37 +29,56 @@ namespace trd
 using ADC_t = std::uint16_t;
 using ArrayADC = std::array<ADC_t, constants::TIMEBINS>;
 
-using TimeStamp = o2::dataformats::TimeStamp<double>;
-
-class Digit : public TimeStamp
+// Digit class for TRD
+// Notes:
+//    Shared pads:
+//        the lowe mcm and rob is chosen for a given shared pad.
+//        this negates the need for need alternate indexing strategies.
+//        it does however mean that if you are trying to go from pad/row to mcm/rob you need to remember to manually do the shared ones.
+//        TODO we could change the get methods to return the value in negaitve to indicate a shared pad, but I feel this is just added complexity? Comments?
+//        if you are going to have to check for negative you may as well check for it being shared.
+class Digit
 {
  public:
   Digit() = default;
   ~Digit() = default;
-  Digit(const int det, const int row, const int pad, const ArrayADC adc, double t)
-    : mDetector(det), mRow(row), mPad(pad), mADC(adc), TimeStamp(t) {}
+  Digit(const int det, const int row, const int pad, const ArrayADC adc);
+  Digit(const int det, const int row, const int pad); // add adc data in a seperate step
+  Digit(const int det, const int rob, const int mcm, const int channel, const ArrayADC adc);
+  Digit(const int det, const int rob, const int mcm, const int channel); // add adc data in a seperate step
+
   // Copy
   Digit(const Digit&) = default;
   // Assignment
   Digit& operator=(const Digit&) = default;
   // Modifiers
+  void setROB(int rob) { mROB = rob; }
+  void setROB(int row, int pad) { mROB = FeeParam::getROBfromPad(row, pad); }
+  void setMCM(int mcm) { mMCM = mcm; }
+  void setMCM(int row, int pad) { mMCM = FeeParam::getMCMfromPad(row, pad); }
+  void setChannel(int channel) { mChannel = channel; }
```

Comment:
Are these setters actually used? I would expect that we always know at construction which pad these digits refer to.

Reply:
I prefer keeping them for completeness; they are harmless.